File: backend-java/micro-drama-common/src/main/java/com/series/common/entity/Result.java
package com.series.common.entity;

import com.series.common.exception.ResponseCode;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

@Data
public class Result<T> implements Serializable {
    private static final long serialVersionUID = 1L;

    // 响应码（0:成功，非0:失败）
    private int code;
    // 响应消息
    private String msg;
    // 响应数据
    private T data;

    /**
     * 服务器时间
     */
    private Date serverTime;
    /**
     * 堆栈信息, 开发阶段返回
     */
    private Object stack;

    private String path;

    public Result(Integer code, T data) {
        this.code = code;
        this.data = data;
    }

    public Result(ResponseCode responseCode, T data) {
        this.code = responseCode.getCode();
        this.msg = responseCode.getMsg();
        this.data = data;
    }

    public Result() {
    }

    public static <T> Result<T> ok() {
        Result<T> result = new Result<>();
        result.setCode(ResponseCode.OK.getCode());
        result.setMsg("成功");
        return result;
    }

    public static <T> Result<T> ok(T data) {
        Result<T> result = new Result<>();
        result.setCode(ResponseCode.OK.getCode());
        result.setMsg("成功");
        result.setData(data);
        return result;
    }

    public static <T> Result<T> error(String msg) {
        Result<T> result = new Result<>();
        result.setCode(500);
        result.setMsg(msg);
        return result;
    }

    /**
     * 错误返回（带自定义状态码和消息）
     */
    public static <T> Result<T> error(int code, String msg) {
        Result<T> result = new Result<>();
        result.setCode(code);
        result.setMsg(msg);
        return result;
    }

    public Result<T> put(String key, Object value) {
        // 若需要扩展 map 结构，可改为 Map 存储，这里简化演示
        throw new UnsupportedOperationException("如需扩展，可修改 Result 为 Map 存储");
    }
}

File: backend-go/micro-drama-video/internal/response/result.go
// Package response 提供统一 HTTP JSON 响应结构。
//
// 与 Java 微服务 com.series.common.entity.Result 字段对齐（code、msg、data、serverTime），
// 前端和 API 网关可共用同一套解析逻辑。
package response

import "time"

// Result 泛型统一响应包装。T 为业务数据类型，成功时放在 data 字段。
//
// code：0 表示成功，非 0 表示失败（与 Java ResponseCode 约定一致时可再细化）。
// msg：人类可读提示。
// data：业务载荷，失败时通常省略。
// serverTime：服务端时间戳（毫秒），便于排查时区/时钟问题。
type Result[T any] struct {
	Code       int    `json:"code"`
	Msg        string `json:"msg"`
	Data       T      `json:"data,omitempty"`
	ServerTime int64  `json:"serverTime"`
}

// OK 构造成功响应，code=0，msg=成功。
func OK[T any](data T) Result[T] {
	return Result[T]{
		Code:       0,
		Msg:        "成功",
		Data:       data,
		ServerTime: time.Now().UnixMilli(),
	}
}

// Fail 构造失败响应，不含 data。
//
// code：HTTP 层常用 400/500，也可与业务错误码表一致。
// msg：返回给前端的错误说明。
func Fail[T any](code int, msg string) Result[T] {
	return Result[T]{
		Code:       code,
		Msg:        msg,
		ServerTime: time.Now().UnixMilli(),
	}
}
